cpp/src/gandiva/projector.cc

```diff
@@ -24,102 +24,88 @@
 
 #include "arrow/util/hash_util.h"
 #include "arrow/util/logging.h"
-
+#include "gandiva/base_cache_key.h"
 #include "gandiva/cache.h"
 #include "gandiva/expr_validator.h"
+#include "gandiva/gandiva_object_cache.h"
 #include "gandiva/llvm_generator.h"
 
 namespace gandiva {
 
-class ProjectorCacheKey {
```

Comment:
any reason why moved this class to projector.h?

Reply:
Because it was giving me some circular dependency error, so to avoid it I moved it to the projector.h. The FilterCacheKey also is in the filter.h, so I thought it was no issue to move ProjectorCacheKey to projector.h